thread.c

```diff
@@ -1153,6 +1160,12 @@ remove_from_join_list(VALUE arg)
 
 static rb_hrtime_t *double2hrtime(rb_hrtime_t *, double);
 
+static int
+thread_finished(rb_thread_t *th)
+{
+    return th->status == THREAD_KILLED || th->value != Qundef;
```

Comment:
`th->status == THREAD_KILLED` is needed?

Reply:
I believe so because there are some cases where the thread can be dead but did not finish.

Reply:
I see.